packages/embarkjs/src/blockchain.js

```diff
@@ -367,6 +375,21 @@ Contract.prototype.send = function(value, unit, _options) {
   return this.blockchainConnector.send(options);
 };
 
+Contract.prototype.deployed = function() {
+  if (this.address === undefined && this._deployPromise === undefined) {
+    throw new Error("contract deployment hasn't happened yet");
+  }
+
+  if (this._deployPromise) {
+    return this._deployPromise;
```

Comment:
This should maybe clean the property up.

Reply:
Can you elaborate?

Reply:
Already do it up on `deploy`.